client/components/account-balances/test/index.test.tsx

```diff
@@ -64,3 +188,116 @@ describe( 'AccountBalancesHeader', () => {
+
+		// Check the available and pending amounts are rendered correctly.
+		expect( availableAmount ).toHaveTextContent( '¥46' );
+		expect( pendingAmount ).toHaveTextContent( '¥123' );
+	} );
+
+	test( 'renders multiple currency tabs', () => {
+		mockGetCurrencyTabTitle.mockImplementation(
+			( currencyCode: string ) => {
+				return `${ currencyCode.toUpperCase() } Balance`;
+			}
+		);
+		mockOverviews( [
+			createMockOverview( 'eur', 7660, 2739 ),
+			createMockOverview( 'usd', 84875, 47941 ),
+			createMockOverview( 'jpy', 2000, 9000 ),
+		] );
+
+		render( <AccountBalancesTabPanel /> );
+
+		// Check the tab titles are rendered correctly.
+		const tabTitles = document.querySelectorAll(
+			'.components-tab-panel__tabs-item'
+		) as NodeListOf< HTMLElement >;
+
+		expect( tabTitles[ 0 ] ).toHaveTextContent( 'EUR Balance' );
+		expect( tabTitles[ 1 ] ).toHaveTextContent( 'USD Balance' );
+		expect( tabTitles[ 2 ] ).toHaveTextContent( 'JPY Balance' );
+
+		const eurAvailableAmount = document.querySelector(
+			'.wcpay-account-available__balance_item .wcpay-account-balances__balances__item__amount'
+		) as HTMLElement;
+
+		const eurPendingAmount = document.querySelector(
+			'.wcpay-account-pending__balance_item .wcpay-account-balances__balances__item__amount'
+		) as HTMLElement;
+
+		// Check the available and pending amounts are rendered correctly for the first tab.
+		expect( eurAvailableAmount ).toHaveTextContent( '€27.39' );
+		expect( eurPendingAmount ).toHaveTextContent( '€76.60' );
+
+		// Create a click event to trigger clicking on the other tabs.
+		const clickEvent = new MouseEvent( 'click', {
+			view: window,
+			bubbles: true,
+			cancelable: true,
+		} );
+
+		// Trigger clicking on the second tab.
+		tabTitles[ 1 ].dispatchEvent( clickEvent );
+
+		const usdAvailableAmount = document.querySelector(
+			'.wcpay-account-available__balance_item .wcpay-account-balances__balances__item__amount'
+		) as HTMLElement;
+
+		const usdPendingAmount = document.querySelector(
+			'.wcpay-account-pending__balance_item .wcpay-account-balances__balances__item__amount'
+		) as HTMLElement;
+
+		// Check the available and pending amounts are rendered correctly for the first tab.
+		expect( usdAvailableAmount ).toHaveTextContent( '$479.41' );
+		expect( usdPendingAmount ).toHaveTextContent( '$848.75' );
+
+		// Trigger clicking on the second tab.
+		tabTitles[ 2 ].dispatchEvent( clickEvent );
+
+		const jpyAvailableAmount = document.querySelector(
+			'.wcpay-account-available__balance_item .wcpay-account-balances__balances__item__amount'
+		) as HTMLElement;
+
+		const jpyPendingAmount = document.querySelector(
+			'.wcpay-account-pending__balance_item .wcpay-account-balances__balances__item__amount'
+		) as HTMLElement;
+
+		// Check the available and pending amounts are rendered correctly for the first tab.
+		expect( jpyAvailableAmount ).toHaveTextContent( '¥90' );
+		expect( jpyPendingAmount ).toHaveTextContent( '¥20' );
+	} );
```

Comment:
For triggering events, I'd suggest can use `fireEvent` from react-testing-library.

In the example below, I'm also using `screen` from RTL, which is like `document` with react-testing-library methods attached.

As above, this test assumes that we've added a title attribute to the `p` element that renders the balance amount in `<BalanceBlock />`.

Here's the same test written with `fireEvent`:

```tsx
import { render, fireEvent, screen } from '@testing-library/react';

	test( 'renders multiple currency tabs', () => {
		mockGetCurrencyTabTitle.mockImplementation(
			( currencyCode: string ) => {
				return `${ currencyCode.toUpperCase() } Balance`;
			}
		);
		mockOverviews( [
			createMockOverview( 'eur', 7660, 2739 ),
			createMockOverview( 'usd', 84875, 47941 ),
			createMockOverview( 'jpy', 2000, 9000 ),
		] );

		render( <AccountBalancesTabPanel /> );

		// Using the `screen` import from @testing-library/react allows us to query the rendered DOM
		// Getting all by 'tab' role will return all tab title elements.
		const tabTitles = screen.getAllByRole( 'tab' );

		expect( tabTitles[ 0 ] ).toHaveTextContent( 'EUR Balance' );
		expect( tabTitles[ 1 ] ).toHaveTextContent( 'USD Balance' );
		expect( tabTitles[ 2 ] ).toHaveTextContent( 'JPY Balance' );

		const eurAvailableAmount = screen.getByTitle( 'Available funds' );
		const eurPendingAmount = screen.getByTitle( 'Pending funds' );

		// Check the available and pending amounts are rendered correctly for the first tab.
		expect( eurAvailableAmount ).toHaveTextContent( '€27.39' );
		expect( eurPendingAmount ).toHaveTextContent( '€76.60' );

		fireEvent.click( tabTitles[ 1 ] );

		const usdAvailableAmount = screen.getByTitle( 'Available funds' );
		const usdPendingAmount = screen.getByTitle( 'Pending funds' );

		// Check the available and pending amounts are rendered correctly for the second tab.
		expect( usdAvailableAmount ).toHaveTextContent( '$479.41' );
		expect( usdPendingAmount ).toHaveTextContent( '$848.75' );

		// Trigger clicking on the second tab.
		fireEvent.click( tabTitles[ 2 ] );

		const jpyAvailableAmount = screen.getByTitle( 'Available funds' );
		const jpyPendingAmount = screen.getByTitle( 'Pending funds' );

		// Check the available and pending amounts are rendered correctly for the third tab.
		expect( jpyAvailableAmount ).toHaveTextContent( '¥90' );
		expect( jpyPendingAmount ).toHaveTextContent( '¥20' );
	} );
```

Reply:
Thanks for the tip. I've updated the tests to use `fireEvent`. That's much cleaner. :)